roundwared/recording_collection.py

```diff
@@ -259,5 +259,8 @@ def _update_playlist_timed(self):
         # Make a list of all assets that aren't banned.
         self.playlist_timed = [item.asset for item in playlist
                                if not self._banned(item.asset)]
+        # filter playlist_timed for current tag filters by comparing
+        # to self.all which has already been filtered by tags
+        self.playlist_timed = set(self.playlist_timed).intersection(self.all)
```

Comment:
Yes, I agree(!), but I tried a few things to that effect initially and wasn't able to get anything to work so I reverted to this two-step and less elegant approach. I'm all ears if you want to make a suggestion?!